js/src/redux/providers/signerMiddleware.js

```diff
@@ -73,33 +74,49 @@ export default class SignerMiddleware {
 
     // Sign request in-browser
     const transaction = payload.sendTransaction || payload.signTransaction;
+
     if (wallet && transaction) {
-      (transaction.nonce.isZero()
+      const noncePromise = transaction.nonce.isZero()
         ? this._api.parity.nextNonce(transaction.from)
-        : Promise.resolve(transaction.nonce)
-      ).then(nonce => {
-        let txData = {
-          to: inHex(transaction.to),
-          nonce: inHex(transaction.nonce.isZero() ? nonce : transaction.nonce),
-          gasPrice: inHex(transaction.gasPrice),
-          gasLimit: inHex(transaction.gas),
-          value: inHex(transaction.value),
-          data: inHex(transaction.data)
-        };
-
-        try {
-          // NOTE: Derving the key takes significant amount of time,
-          // make sure to display some kind of "in-progress" state.
-          const signer = Wallet.fromJson(wallet, password);
-          const rawTx = signer.signTransaction(txData);
-
-          handlePromise(this._api.signer.confirmRequestRaw(id, rawTx));
-        } catch (error) {
-          console.error(error);
+        : Promise.resolve(transaction.nonce);
+
+      const { worker } = store.getState().worker;
+
+      const signerPromise = worker && worker._worker.state === 'activated'
```

Comment:
Because it also have an `error` field that gets displayed in Contract Dev. Could get rid of it though